src/Http/Shared/CookieHeaderParserShared.cs

```diff
@@ -28,8 +28,15 @@ public static bool TryParseValues(StringValues values, IDictionary<string, strin
             {
                 if (TryParseValue(value, ref index, supportsMultipleValues, out var parsedName, out var parsedValue))
                 {
+                    if (parsedName == null || string.IsNullOrEmpty(parsedName.Value.Value)
+                        || parsedValue == null || string.IsNullOrEmpty(parsedValue.Value.Value))
```

Comment:
Sorry, just realized this was allocating strings twice.
```suggestion
                    if (parsedName == null || StringSegment.IsNullOrEmpty(parsedName.Value)
                        || parsedValue == null || StringSegment.IsNullOrEmpty(parsedValue.Value))
```